x-pack/solutions/search/plugins/enterprise_search/server/deprecations/index.ts

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {GetDeprecationsContext, RegisterDeprecationsConfig} from "@kbn/core-deprecations-server";
+import {DeprecationsDetails} from "@kbn/core-deprecations-common";
+import { getEnterpriseSearchNodeDeprecation } from './remove_enterprise_search';
+
+export const getRegisteredDeprecations = (): RegisterDeprecationsConfig => {
+  return {
+    getDeprecations: async (ctx: GetDeprecationsContext) => {
+      const deprecations : DeprecationsDetails[] = [];
+      deprecations.push(await getEnterpriseSearchNodeDeprecation(ctx))
+      return deprecations;
```

Comment:
Nitpick: you can just inline the getEnterpriseSearchNodeDeprecation function, unless I'm missing something it doesn't actually do anything async, and isn't very likely to in the future either. The contract doesn't need it to be async either.

```suggestion
      return {
    level: 'critical',
    deprecationType: 'feature',
    title: 'Enterprise Search host(s) must be removed',
    message: 'Enterprise Search is not supported in versions >= 9.x',
    documentationUrl: 'https://docs.elastic.co', //TODO
    correctiveActions: {
      manualSteps: ['Remove all Enterprise Search nodes from your deployment'],
    },
  };
```

Reply:
I'm intending to add 5 or so deprecations here, and was thinking to keep all of their logic separate. I was thinking to follow https://github.com/elastic/kibana/tree/d0d2032f18a37e4c458a26d92092665453b737b0/x-pack/plugins/reporting/server/deprecations as a pattern. WDYT?

Reply:
After talking with Sander offline, I'll just try to hit a good balance of few files, but still good modularization.